.github/pull_request_template.md

```diff
@@ -0,0 +1,54 @@
+Release Notes
+---------
+<!-- If this PR introduces any user-facing changes, document them below as a summary. Delete unused section titles and placeholders. Match the style of previous release notes: https://github.com/confluentinc/terraform-provider-confluent/releases -->
+
+New Features
+- [Briefly describe new features introduced in this PR].
+
+Bug Fixes
+- [Briefly describe any bugs fixed in this PR].
+
+Examples
+- [Briefly describe any Terraform configuration example updates in this PR].
+
+Checklist
+---------
+<!-- 
+Check each item in the checklist to ensure high-quality Terraform development practices are followed. PR approval won't be granted until the checklist is carefully reviewed. Use common sense when checking items: for example, skip items 1–4 if the PR includes documentation updates only. 
+-->
+- [ ] I can successfully build and use a custom Terraform provider binary for Confluent.
+- [ ] I have verified my PR with real Confluent Cloud resources in a production environment.
+- [ ] I have attached manual Terraform verification results or screenshots in the "Test & Review" section below.
+- [ ] I have included appropriate Terraform acceptance tests for any new resources, data sources, or functionality.
+- [ ] I confirm there will be no Terraform drift or backward compatibility issues caused by this PR.
+- [ ] I have updated the corresponding documentation and include relevant examples for this PR.
+
+What
+----
+<!--
+Briefly describe **what** you have changed and **why** these changes are necessary.
+Optionally include: 
+- The problem being solved or the feature being added. 
+- The implementation strategy or approach taken. 
+- Key technical details, design decisions, or any additional context reviewers should be aware of.
+-->
+
+References
+----------
+<!-- Include links to relevant resources for this PR, such as: 
+- Related GitHub issues 
+- Tickets (JIRA, etc.) 
+- Internal documentation or design specs 
+- Other related PRs 
+Copy and paste the links below for easy reference.
+-->
+
+Test & Review
+-------------
+<!-- Has this PR been tested? If so, explain **how** it was tested. Include: 
+- Steps taken to verify the changes. 
+- Links to manual verification documents, logs, or screenshots to save reviewers' time. 
+- Any additional notes on testing (e.g., environments used, edge cases tested). 
+- Screenshot showing successful resource creation.
+Example: - [Manual Verification Document](https://docs.google.com/document/d/1dutVZmbEwJBBqMzx57uCXqllV1SEr2vxnjUrtTPCwBk/edit?tab=t.0#heading=h.6zajc95mev5j)
```

Comment:
I think it might be good to also add a checklist, or at least an expanded description, detailing specific steps that need to be tested.

Ideally, we want plan and apply for all of the operations (Create, Update, Delete, Import), and we also want plan after each apply, to check that we aren't experiencing drift (a plan not being up to date right after an apply).

Reply:
Sure, as we discussed, I will have the link of detailed instruction as part of the MD file.